modApiServer/test/org/aion/api/server/pb/ApiAion0Tests.java

```diff
@@ -0,0 +1,2137 @@
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+
+package org.aion.api.server.pb;
+
+import com.google.protobuf.ByteString;
+import org.aion.api.server.ApiUtil;
+import org.aion.base.type.Address;
+import org.aion.base.util.TypeConverter;
+import org.aion.crypto.ed25519.ECKeyEd25519;
+import org.aion.equihash.EquihashMiner;
+import org.aion.mcf.account.AccountManager;
+import org.aion.mcf.account.Keystore;
+import org.aion.zero.impl.Version;
+import org.aion.zero.impl.blockchain.AionImpl;
+import org.aion.zero.impl.config.CfgAion;
+import org.aion.zero.impl.db.AionRepositoryImpl;
+import org.aion.zero.impl.types.AionBlock;
+import org.aion.zero.impl.types.AionBlockSummary;
+import org.aion.zero.types.AionTransaction;
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+public class ApiAion0Tests {
+
+    private byte[] msg, socketId, hash;
+
+    private static final int MSG_HASH_LEN = 8;
+    private static final int RSP_HEADER_NOHASH_LEN = 3;
+    private static final int REQ_HEADER_NOHASH_LEN = 4;
+    private static final int RSP_HEADER_LEN = RSP_HEADER_NOHASH_LEN + MSG_HASH_LEN;
+
+    public ApiAion0Tests() {
+        msg = "test message".getBytes();
+        socketId = RandomUtils.nextBytes(5);
+        hash = RandomUtils.nextBytes(ApiUtil.HASH_LEN);
+    }
+
+    private byte[] stripHeader(byte[] rsp) {
+        boolean hasHash = (rsp[2] == 1);
+        int bodyLen = rsp.length - (hasHash ? RSP_HEADER_LEN : RSP_HEADER_NOHASH_LEN);
+
+        if (hasHash) {
+            return Arrays.copyOfRange(rsp, RSP_HEADER_LEN, RSP_HEADER_LEN + bodyLen);
+        } else {
+            return Arrays.copyOfRange(rsp, RSP_HEADER_NOHASH_LEN, RSP_HEADER_NOHASH_LEN + bodyLen);
+        }
+    }
+
+    @Test
+    public void TestCreate() {
+        System.out.println("run TestCreate.");
+        AionImpl impl = AionImpl.inst();
+        ApiAion0 api = new ApiAion0(impl);
+        api.shutDown();
+    }
+
+    @Test
+    public void TestHeartBeatMsg() {
+        System.out.println("run TestHeartBeatMsg.");
+        AionImpl impl = AionImpl.inst();
+        ApiAion0 api = new ApiAion0(impl);
+
+        byte[] msg = ByteBuffer.allocate(api.getApiHeaderLen()).put(api.getApiVersion())
+                .put((byte) Message.Servs.s_hb_VALUE).array();
+        assertTrue(ApiAion0.heartBeatMsg(msg));
+        assertFalse(ApiAion0.heartBeatMsg(null));
+
+        msg[0] = 0;
+        assertFalse(ApiAion0.heartBeatMsg(msg));
+
+        api.shutDown();
+    }
+
+    @Test
+    public void TestProcessProtocolVersion() {
+        System.out.println("run TestProcessProtocolVersion.");
+        AionImpl impl = AionImpl.inst();
+        ApiAion0 api = new ApiAion0(impl);
+
+        byte[] request = ByteBuffer.allocate(msg.length + REQ_HEADER_NOHASH_LEN + hash.length).put(api.getApiVersion())
```

Comment:
Looks like this block of code repeats enough in other tests that it's worth extracting into a helper method

Reply:
done